src/test/java/com/squareup/javapoet/TypeSpecTest.java

```diff
@@ -1043,6 +1043,32 @@ public void interfaceDefaultMethods() throws Exception {
         + "  }\n"
         + "}\n");
   }
+  
+  @Test public void indexedElseIf() throws Exception {
+    TypeSpec taco = TypeSpec.classBuilder("Taco")
+        .addMethod(MethodSpec.methodBuilder("choices")
+            .beginControlFlow("if ($1L != null || $1L == $2L)", "mTaco", "mOtherTaco")
```

Comment:
mind dropping the `m` prefix here? We don't want to endorse that code style, not even in generated code!